modules/3d/src/pointcloud/io_ply.cpp

```diff
@@ -426,18 +504,18 @@ void PlyDecoder::parseBody(std::ifstream &file, std::vector<Point3f> &points, st
             size_t offset = vertexOffsets[j];
             if (offset != (size_t)(-1))
             {
-                switch (p.valType)
+                if (colorKeys.count(p.name) > 0)
                 {
-                case CV_8U: case CV_8S:
-                    *(vertexData.bytes.data() + offset) = (uchar)ival;
-                    break;
-                case CV_32F:
-                    *(float*)(vertexData.bytes.data() + offset) = fval;
-                    break;
-                default:
-                    // the rest are unused
-                    break;
+                    switch (p.valType)
+                    {
+                    case  CV_8U: case  CV_8S: fval = ival / 255.f;   break;
+                    case CV_16U: case CV_16S: fval = ival / 65535.f; break;
+                    case CV_32U: case CV_32S: fval = (float)ival; break;
```

Comment:
Why 32-bit int is not normalized? What about float branch present in the old version of the code?